cordapp-cpk2/src/main/kotlin/net/corda/plugins/cpk2/OsgiExtension.kt

```diff
@@ -188,6 +188,11 @@ open class OsgiExtension(objects: ObjectFactory, jar: Jar) {
 
     @get:Input
     val exports: Provider<String> = _exports.map { names ->
+
+        require(names.none { it.startsWith("net.corda.") }) {
+            "CorDapps must not export \"net.corda\" package"
+        }
+
```

Comment:
This only catches the packages that either the plugin computes when
```gradle
osgi {
    autoExport = true
}
```
or that the CPK authors explicitly export themselves. However, if you use `autoExport = false` then the plugin delegates this calculation to Bnd and the `@Export` annotations etc. This allows CPK authors who understand OSGi to declare some of the packages in their "main" jar as "Private-Package", i.e. invisible from other "main" jars in the sandbox.

You will probably want to perform this kind of check in the `VerifyBundle` task instead, adding extra errors into `verifier`:
```kotlin
verifier.error("Export Package clause found for Corda package [%s]", packageName)
```
Any violations detected are reported together at the end.

Reply:
I've moved the check into VerifyBundle and used verifier.error for every net.corda package found.
Also expanded the existing test to check two net.corda packages are detected.